typescript-test-samples/launchdarkly-lambda-dynamodb/README.md

```diff
@@ -0,0 +1,97 @@
+
+You will need a LaunchDarkly account. If you don't have one yet, you can [start a free trial](https://app.launchdarkly.com/signup). Once you have an account, you'll need your server-side SDK key. Your environment's SDK key is available in the Projects tab of your [Account settings](https://app.launchdarkly.com/settings/projects) page. Please replace the `LAUNCHDARKLY_SDK_KEY` environment variable in [`cdk-stack.js`](cdk/lib/cdk-stack.js) prior to deploying the stack.
+
+The example Lambda functions look for a boolean flag called `new-feature`. You will need to create that flag to try the functions as is or change the key in the function to match a flag within your account.
+
+For a complete guide to using LaunchDarkly in AWS serverless, including how to set up the DynamoDB store and/or the Relay Proxy within your AWS environment and how to ensure that all events are captured by the SDKs within an ephemeral Lambda function, visit [this guide](https://launchdarkly.com/blog/using-launchdarkly-in-aws-serverless/).
+
+The project also includes the code necessary to build and deploy a [Lambda Layer](https://docs.aws.amazon.com/lambda/latest/dg/chapter-layers.html) to easily add the necessary LaunchDarkly dependencies to any Lambda function that requires them. More on how to create and use the Layer below.
+
+## The LaunchDarkly SDKs
+
+Adding LaunchDarkly feature flags to any Lambda function always begins with installing the appropriate SDK for the Lambda runtime you are using. You can find a [full list of server-side SDKs here](https://docs.launchdarkly.com/sdk). For the Node.js runtime, that would mean using the [Node Server SDK](https://docs.launchdarkly.com/sdk/server-side/node-js) via NPM.
+
+```bash
+npm install launchdarkly-node-server-sdk
+```
+
+If you would like to use [DynamoDB as a persistent feature store](https://docs.launchdarkly.com/sdk/features/storing-data/dynamodb/?q=dynamo), you'll need to also install those dependencies.
+
+```bash
+npm install launchdarkly-node-server-sdk-dynamodb
+```
+
+This works well for a single Lambda function, but, if you are using feature flags across a number of Lambda functions, the easiest method to maintain the depencies across all of them is using a Lambda Layer. The code provided here will assist you in creating the Layer.
+
+### Creating/Updating the Lambda Layer
+
+The necessary dependencies are already included in this project under `/cdk/resources/layers`. It contains both the Node.js SDK and the DynamoDB extension. In order to ensure that this asset has the latest versions of these SDKs, we've included a script that will download the latest versions and archive them into the proper folder. It is recommended that you run this before deploying any of the assets using the provided CDK scripts. All you need to do is:
+
+```bash
+cd  create-layer
+npm run build
```

Comment:
Small update... users need to run `npm install` in this directory before running `npm run build`.

Otherwise they get:

```
node:internal/modules/cjs/loader:1152
  throw err;
  ^

Error: Cannot find module 'archiver'
Require stack:
- /Users/.../serverless-test-samples/typescript-test-samples/launchdarkly-lambda-dynamodb/create-layer/index.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1149:15)
    at Module._load (node:internal/modules/cjs/loader:990:27)
    at Module.require (node:internal/modules/cjs/loader:1237:19)
    at require (node:internal/modules/helpers:176:18)
    at Object.<anonymous> (/Users/.../serverless-test-samples/typescript-test-samples/launchdarkly-lambda-dynamodb/create-layer/index.js:3:18)
    at Module._compile (node:internal/modules/cjs/loader:1378:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1437:10)
    at Module.load (node:internal/modules/cjs/loader:1212:32)
    at Module._load (node:internal/modules/cjs/loader:1028:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:142:12) {
  code: 'MODULE_NOT_FOUND',
```

Reply:
Also, not getting `cdk/resources/layers/layer.zip` created when I run `npm run build`. Not sure what's going on - maybe an async issue? Seems like the script runs and ends early without waiting for the archive file to be created.

Reply:
Looks like its failing because `cdk/resources/layers` doesn't exist.

Please either create the directory and use a `.gitkeep` file to check it in, or have the build script create it if it doesn't already exist.

I created the directory manually, and the zip file was created as expected.